### PR TITLE
Clean custom OSL extensions, unneeded comments

### PR DESCRIPTION
**Pull request by luisbarrancos**

### CODE REVIEW

File: src/appleseed.shaders/as_osl_extensions.h.in [hunk 1]
```diff
@@ -30,20 +30,17 @@
...
-#define APPLESEED_VERSION_MAJOR     ${appleseed_version_major}
-#define APPLESEED_VERSION_MINOR     ${appleseed_version_minor}
-#define APPLESEED_VERSION_PATCH     ${appleseed_version_patch}
+#define APPLESEED_VERSION_MAJOR     1
+#define APPLESEED_VERSION_MINOR     6
+#define APPLESEED_VERSION_PATH      0
 
 #define APPLESEED_VERSION               \
-    APPLESEED_VERSION_MAJOR * 10000 +   \
+    APPLESEED_VERSION_MAJOR + 10000 +   \
```
Comment (dictoon) [inline src/appleseed.shaders/as_osl_extensions.h.in]: That doesn't look right.

Comment (luisbarrancos) [inline src/appleseed.shaders/as_osl_extensions.h.in]: @dictoon oops, that does look interesting though

Comment (est77) [inline src/appleseed.shaders/as_osl_extensions.h.in]: This file is not actually a final file but a cmake "template" file. cmake replaces the ${appleseed_version_xxx} tags by the right values and generates the actual as_osl_extensions.h file as part of the build.

We should not remove the tags and hardcode the version number or change the APPLESEED_VERSION macro.
The other changes in the PR are good in my opinion.

Comment (luisbarrancos) [inline src/appleseed.shaders/as_osl_extensions.h.in]: Adding back the ${foobar} tags

File: src/appleseed.shaders/as_osl_extensions.h.in [hunk 2]
```diff
@@ -30,20 +30,17 @@
...
-
-// appleseed version
 #define APPLESEED_VERSION_MAJOR     ${appleseed_version_major}
 #define APPLESEED_VERSION_MINOR     ${appleseed_version_minor}
-#define APPLESEED_VERSION_PATCH     ${appleseed_version_patch}
+#define APPLESEED_VERSION_PATH      ${appleseed_version_patch}
 
 #define APPLESEED_VERSION               \
-    APPLESEED_VERSION_MAJOR * 10000 +   \
+    APPLESEED_VERSION_MAJOR + 10000 +   \
```
Comment (dictoon) [inline src/appleseed.shaders/as_osl_extensions.h.in]: I really don't think that makes any sense. The multiplication was correct.

File: src/appleseed.shaders/as_osl_extensions.h.in [hunk 3]
```diff
@@ -30,20 +30,17 @@
 #ifndef AS_OSL_EXTENSIONS_H
 #define AS_OSL_EXTENSIONS_H
 
-///////////////////////////////////////////////////////////////////////////////
-
-// appleseed version
 #define APPLESEED_VERSION_MAJOR     ${appleseed_version_major}
 #define APPLESEED_VERSION_MINOR     ${appleseed_version_minor}
-#define APPLESEED_VERSION_PATCH     ${appleseed_version_patch}
+#define APPLESEED_VERSION_PATH      ${appleseed_version_patch}
```
Comment (dictoon) [inline src/appleseed.shaders/as_osl_extensions.h.in]: That's a regression, the previous name (`APPLESEED_VERSION_PATCH`) was correct.

File: src/appleseed.shaders/as_osl_extensions.h.in [hunk 4]
```diff
@@ -202,3 +170,4 @@ closure color as_disney(
     float   clearcoat_gloss) BUILTIN;
 
 #endif // AS_OSL_EXTENSIONS_H
+
```
Comment (dictoon) [inline src/appleseed.shaders/as_osl_extensions.h.in]: Remove redundant blank line.

Comment (luisbarrancos) [inline src/appleseed.shaders/as_osl_extensions.h.in]: @dictoon should i remove the last lines of the OSL shaders as well?

Comment (dictoon) [inline src/appleseed.shaders/as_osl_extensions.h.in]: Thanks! Yes, please.

File: src/appleseed.shaders/as_osl_extensions.h.in (hunk 2 above)
Comment (luisbarrancos) [inline src/appleseed.shaders/as_osl_extensions.h.in]: @dictoon yes it was, disregard

**Comment by dictoon** [comment]
Would make sense to squash the commits into one before merging.

File: src/appleseed.shaders/as_osl_extensions.h.in (hunk 4 above)
Comment (luisbarrancos) [inline src/appleseed.shaders/as_osl_extensions.h.in]: @dictoon ok, will do so on its own PR
